test/e2e/framework/util.go

```diff
@@ -4095,6 +4095,18 @@ func CheckForControllerManagerHealthy(duration time.Duration) error {
 	return nil
 }
 
+// Returns number of registered Nodes excluding Master Node.
+func NumberOfNodes(c clientset.Interface) (int, error) {
```

Comment:
I think there should be a function like this somewhere already IIRC. And that one should also have retry-logic. Can you please check for it?

Reply:
I was looking for it and there isn't.

About retries - we should probably fix it, but for now it's consistent with a number of functions underneath doing similar stuff. We should fix the all together.

Reply:
You can probably re-use `waitListSchedulableNodesOrDie()`.

Reply:
See above ^

Reply:
Good finding. Done.

Reply:
Also maybe rename this function to `NumberOfNodesRegistered` or sth - to make it clearer what it's giving?